### PR TITLE
NAS-101877 / 11.3 / feat(middlewared/update): implement `update.set_train`

### PR DESCRIPTION
**Pull request by william-gr**
Ticket:	NAS-101877

### CODE REVIEW

File: src/middlewared/middlewared/plugins/update.py [hunk 1]
```diff
@@ -268,6 +268,23 @@ def get_trains(self):
             'selected': selected,
         }
 
+    @accepts(Str('train'))
+    def set_train(self, train):
+        """
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/update.py]: Don't we want any check here? Like in `update`

Comment (william-gr) [reply src/middlewared/middlewared/plugins/update.py]: We probably do, good thought

Comment (themylogin) [reply src/middlewared/middlewared/plugins/update.py]: What is a point of `call("update.set_train", null)`?

**Comment by themylogin** [comment]
What is a point of `call("update.set_train", null)`?

**Comment by william-gr** [comment]
@themylogin return get_trains, otherwise we would need to make two calls of these, which depend on network

**Comment by themylogin** [comment]
@william-gr have not thought of that. But things like these tend to complicate over time, and next time we'll need to pack more actions into single call will lead to even more strange code. I don't insist on doing anything, but in cases like this I would prefer to have simple and clean public API and hide ugliness in a wrapper method:
```
@accepts(Str("train", required=True, empty=False))
def set_train(self, train):
    ...

@private
def handle_train_change(self, new_train):
    if new_train:
        self.set_train()
    return self.get_trains()
```
Also is number of calls (furthermore, non-repeated) an important thing in TN HA? I've thought we have some kind of fiber optic between them, not 3G :)

**Comment by themylogin** [comment]
If I (as e.g. a new UI developer) will start calling `set_train(null)` by mistake and it won't raise any error and just silently don't do anything, I will spend more time trying to investigate what's goind on.

**Comment by william-gr** [comment]
@themylogin that doesnt quite work, set_train needs to call get_trains, and update needs to check for the selected train get_trains returns.

As far as number of calls, this has nothing to do with TN HA, its network call, external, to the internet.

**Comment by william-gr** [comment]
But yes, I get your point, I will rework on that.

File: src/middlewared/middlewared/plugins/update.py [hunk 2]
```diff
@@ -268,6 +268,58 @@ def get_trains(self):
             'selected': selected,
         }
 
+    @accepts(Str('train'))
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/update.py]: `empty=False`?

**Pull request merged**